cypress/e2e/buy-shirt.cy.ts

```diff
@@ -1,23 +1,35 @@
+import {AddressStepPage, LoginPage, MenuContentPage, PaymentStepPage} from "../page/index";
+import {ProductsListPage, ShippingStepPage, ShoppingCartPage} from "../page/index";
+
+const menuContentPage = new MenuContentPage();
+const productsListPage = new ProductsListPage();
+const shoppingCartPage = new ShoppingCartPage();
+const loginPage = new LoginPage();
+const addressStepPage = new AddressStepPage();
+const shippingStepPage = new ShippingStepPage();
+const paymentStepPage = new PaymentStepPage();
+
 describe("Buy a t-shirt", () => {
   it("then the t-shirt should be bought", () => {
-    cy.visit("https://automationpractice.com/");
-    cy.get("#block_top_menu > ul > li:nth-child(3) > a").click();
-    cy.get("#center_column a.button.ajax_add_to_cart_button.btn.btn-default").click();
-    cy.get("[style*='display: block;'] .button-container > a").click();
-    cy.get(".cart_navigation span").click();
+    menuContentPage.visitMenuContentPage();
+    menuContentPage.goToTShirtMenu();
+
+    productsListPage.addToCart();
+    productsListPage.goToShoppingCart();
 
-    cy.get("#email").type("aperdomobo@gmail.com");
-    cy.get("#passwd").type("WorkshopProtractor");
+    shoppingCartPage.goToLogin();
 
-    cy.get("#SubmitLogin > span").click();
+    loginPage.typeEmail();
+    loginPage.typePassword();
+    loginPage.goToAddress();
 
-    cy.get(".cart_navigation > .button > span").click();
-    cy.get("#cgv").click();
+    addressStepPage.goToShipping();
 
-    cy.get(".cart_navigation > .button > span").click();
-    cy.get(".bankwire").click();
+    shippingStepPage.acceptTerms();
+    shippingStepPage.goToPayment();
 
-    cy.get("#cart_navigation > .button > span").click();
+    paymentStepPage.selecBankWireOption();
+    paymentStepPage.buyTShirt();
     cy.get("#center_column > div > p > strong")
         .should("have.text", "Your order on My Store is complete.");
```

Comment:
@AlejaGonzalezV No Aleja, Holmes have shared that its a good practice to assert in the test file, so unless you have like 5 or 6 assertions its better to keep the validations on the test file 👅

Reply:
I have reduced my strictness, validation in page object is fine, if the following requirements are followed:

1. name should be clear something like: `validateConfirmationMessage` or `assertConfirmationMessage`
2. the message to validate should be received as parameter.

I continue thinking that it is better outside of the Page Objects, because of the single responsibility principle. The old-school where I come from, says that page objects should not have assertions:
https://stackoverflow.com/a/11577828

Also, Martin Fowler recommends the [same that you say](https://martinfowler.com/bliki/PageObject.html#:~:text=Page%20objects%20are%20commonly%20used,carry%20out%20the%20assertion%20logic.):
"Page objects are commonly used for testing, but should not make assertions themselves. Their responsibility is to provide access to the state of the underlying page. It's up to test clients to carry out the assertion logic."

So, the best practice is to not perform assertions or validations on page objects, but, in some cases (educational) we can be flexible :)